### PR TITLE
Add missing gettext annotations

### PR DESCRIPTION
**Pull request by mzazrivec**
@himdel review please?

### CODE REVIEW

File: src/dialog-editor/components/modal/modalComponent.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ class ModalController {
           };
           const titleLabel = this.modalData.type in titles &&
             titles[this.modalData.type];
-          this.modalTitle =  __(`Edit ${titleLabel} Field`);
+          this.modalTitle =  sprintf(__("Edit %s Field"), titleLabel);
```
Comment (mzazrivec) [inline src/dialog-editor/components/modal/modalComponent.ts]: I'm guessing we have to add `sprintf()` somewhere for this to work, am I correct @himdel?

**Comment by mzazrivec** [comment]
Maybe this one should be merged before https://github.com/ManageIQ/ui-components/pull/320 and that one should be rebased afterwards? (it contains gettext catalogs with the strings).

File: src/dialog-editor/components/modal/modalComponent.ts (hunk 1 above)
Comment (himdel) [reply src/dialog-editor/components/modal/modalComponent.ts]: Looks like both ui-classic and ui-service depend on sprintf and provide it on window.

So only the demo might fail on missing sprintf..

But `yarn add sprintf-js` plus:

```diff
diff --git a/src/vendor.ts b/src/vendor.ts
index 593d7cc..772ac71 100644
--- a/src/vendor.ts
+++ b/src/vendor.ts
@@ -17,3 +17,4 @@ import 'ui-select';
 import 'patternfly-bootstrap-treeview';
 import * as numeral from 'numeral';
 window['numeral'] = numeral;
+window['sprintf'] = require('sprintf-js').sprintf;
```

fixes that :)

**Comment by himdel** [comment]
LGTM, but we should add that sprintf thing..

File: src/dialog-editor/components/modal/modalComponent.ts (hunk 1 above)
Comment (mzazrivec) [reply src/dialog-editor/components/modal/modalComponent.ts]: Done.

**Comment by himdel** [comment]
Backported to gaprindashvili:

```
commit 176830a1e1b7950764cae81628da081d3b36a3b1
Author: Martin Hradil <himdel@seznam.cz>
Date:   Wed Jul 18 15:06:56 2018 +0100

    Merge pull request #322 from mzazrivec/add_missing_gettext_annotations
    
    Add missing gettext annotations
    
    (cherry picked from commit 49d0c16bd2a4303444fa7112e206a84a3f8af22f)
```

@miq-bot add_label gaprindashvili/backported
@miq-bot remove_label gaprindashvili/yes

**Comment by ZitaNemeckova** [comment]
Released in v1.0.33
https://github.com/ManageIQ/manageiq-ui-service/pull/1464
https://github.com/ManageIQ/manageiq-ui-classic/pull/4352

**Pull request merged**